packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -121,58 +121,73 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
+      const visitStartDatetime = new Date(data.visitStartDate).setHours(visitStartHours, visitStartMinutes);
+      return new Date(visitStartDatetime) <= new Date();
+    };
+
+    return z
+      .object({
+        visitStartDate: z.date().refine(
+          (value) => {
+            const today = dayjs();
+            const startDate = dayjs(value);
+            return displayVisitStopDateTimeFields ? true : startDate.isSameOrBefore(today, 'day');
           },
+          t('invalidVisitStartDate', 'Start date needs to be on or before {{firstEncounterDatetime}}', {
+            firstEncounterDatetime: formatDatetime(new Date()),
+            interpolation: {
+              escapeValue: false,
+            },
+          }),
+        ),
+        visitStartTime: z
+          .string()
+          .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format')),
+        visitStartTimeFormat: z.enum(['PM', 'AM']),
+        visitStopDate: displayVisitStopDateTimeFields ? z.date() : z.date().optional(),
+        visitStopTime: displayVisitStopDateTimeFields
+          ? z
+              .string()
+              .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format'))
+          : z.string().optional(),
+        visitStopTimeFormat: displayVisitStopDateTimeFields ? z.enum(['PM', 'AM']) : z.enum(['PM', 'AM']).optional(),
+        programType: z.string().optional(),
+        visitType: z.string().refine((value) => !!value, t('visitTypeRequired', 'Visit type is required')),
+        visitLocation: z.object({
+          display: z.string(),
+          uuid: z.string(),
         }),
-      ),
-      visitStartTime: z
-        .string()
-        .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format')),
-      visitStartTimeFormat: z.enum(['PM', 'AM']),
-      visitStopDate: displayVisitStopDateTimeFields ? z.date() : z.date().optional(),
-      visitStopTime: displayVisitStopDateTimeFields
-        ? z
-            .string()
-            .refine((value) => value.match(time12HourFormatRegex), t('invalidTimeFormat', 'Invalid time format'))
-        : z.string().optional(),
-      visitStopTimeFormat: displayVisitStopDateTimeFields ? z.enum(['PM', 'AM']) : z.enum(['PM', 'AM']).optional(),
-      programType: z.string().optional(),
-      visitType: z.string().refine((value) => !!value, t('visitTypeRequired', 'Visit type is required')),
-      visitLocation: z.object({
-        display: z.string(),
-        uuid: z.string(),
-      }),
-      visitAttributes: z.object(visitAttributes),
-    });
+        visitAttributes: z.object(visitAttributes),
+      })
+      .refine((data) => validateStartTime(data), {
+        message: t('futureStartTime', 'Visit start time cannot be in the future'),
+        path: ['visitStartTime'],
+      });
```

Comment:
This is the key change. It runs whatever value gets filled into the `Visit start time` input against `validateStartTime`. If the start time is in the future, the validation error message above is rendered.